### DISCUSSION TITLE
VSCode reports to LSP that it has no capabilities for Semantic Tokens

### DISCUSSION THREAD

**Discussion by dvoyni** [microsoft/vscode-discussions#1060] [Extension Development QnA] [answered]
Hello. I'm implementing LSP for my language. So far I succeeded with document sync,  references provider, definition provider and some others. Now I want to implement semantic tokens provider, but on server initialisation request I receive in capabilities, that vscode does not support it (most of rest capabilities are true):
```json
"semanticTokens": {
  "dynamicRegistration": false,
  "requests": {
    "range": null,
    "full": null
  },
  "tokenTypes": null,
  "tokenModifiers": null,
  "formats": null,
  "overlappingTokenSupport": false,
  "multilineTokenSupport": false,
  "serverCancelSupport": false,
  "augmentsSyntaxTokens": false,
}
```

And, obviously vscode sends neither `textDocument/semanticTokens/range` nor `textDocument/semanticTokens/full` even if I notify it that server has support:
```json
"semanticTokensProvider": {
  "legend": {
    "tokenTypes": ["type", "class", ...],
    "tokenModifiers": [...],
  },
  "range": true,
  "full": {
    "delta": false,
  }
}
```

What have I tried:
1. Google
2. Tried different color themes (including Dark+, Light+ which should support semantic colours)
3. Added `"editor.semanticHighlighting.enabled": true` in `settings.json`
4. Added in my plugin's `package.json`: 
```json
  "contributes": {
    "configurationDefaults": {
      "[nar]": {
        "editor.semanticHighlighting.enabled": true
      }
    },   
    "languages": [
      {
        "id": "nar",
        ...
``` 

Plugin is pretty simple, it just starts executable with LSP and proxies requests between vscode and LSP. Sources are here: https://github.com/nar-lang/vscode

Any ideas what do I miss?

**Comment by isidorn** [comment]
@aeschli might be able to help here

**Reply by aeschli** [reply]
@dbaeumer ...

**Comment by dbaeumer** [comment, marked as answer]
Your language-client is pretty old (e.g. https://github.com/nar-lang/vscode/blob/main/client/package.json#L17). Can you please update to latest (e.g. 9.0.1)

**Reply by dvoyni** [reply]
Thank you, that helped. Looks like I took the code from one of outdated samples. Sorry for bothering
